Python/bytecodes.c

```diff
@@ -2234,12 +2234,12 @@ dummy_func(
             PyDict_WatchEvent event = old_value == NULL ? PyDict_EVENT_ADDED : PyDict_EVENT_MODIFIED;
             new_version = _PyDict_NotifyEvent(tstate->interp, event, dict, name, PyStackRef_AsPyObjectBorrow(value));
```

Comment:
I think we should structure this to match the ordering in `dictobject.c`:

https://github.com/python/cpython/blob/b6d0a40a42fa17d087e502245b29cde46368ac07/Objects/dictobject.c#L1693-L1707

1) Update GC tracking if necessary
2) Notify event
3) Store value and version tag
4) Decref old value